### PR TITLE
Version 2

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

File: manifest.toml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+packaging_format = 2
+
+id = "redirect"
+name = "Redirect"
+description.en = "Create a redirection or a proxy to another path"
+description.fr = "Créer une redirection ou un proxy vers un autre emplacement"
+
+version = "1.0.2~ynh1"
+
+maintainers = ["alexAubin"]
```
Comment (alexAubin) [inline manifest.toml]:
```suggestion
maintainers = []
```

:face_in_clouds:

File: manifest.toml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+[upstream]
+license = "AGPL-3.0-or-later"
+website = "https://github.com/YunoHost-Apps/redirect_ynh"
+
+[integration]
+yunohost = ">= 11.2"
+architectures = "all"
+multi_instance = true
+ldap = false
+sso = false
```
Comment (alexAubin) [inline manifest.toml]:
```suggestion
ldap = "not_relevant"
sso = "not_relevant"
```

File: manifest.toml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    [install.path]
+    type = "path"
+    default = "/redirect"
+
+    [install.redirect_path]
+    ask.en = "Redirect destination path"
+    ask.fr = "Emplacement de destination"
+    type = "string"
+    example = "http://127.0.0.1:8080/app/"
+    default = "http://127.0.0.1"
```
Comment (alexAubin) [inline manifest.toml]:
```suggestion
```

it feels weird to have a default value for this ...

File: scripts/install [hunk 4]
```diff
@@ -8,27 +8,13 @@
...
 #=================================================
 # RETRIEVE ARGUMENTS FROM THE MANIFEST
 #=================================================
 
-# Retrieve arguments
-app=$YNH_APP_INSTANCE_NAME
-domain=$YNH_APP_ARG_DOMAIN
-path_url=$YNH_APP_ARG_PATH
 redirect_type=$YNH_APP_ARG_REDIRECT_TYPE
 redirect_path=$YNH_APP_ARG_REDIRECT_PATH
```
Comment (alexAubin) [inline scripts/install]:
```suggestion
```

File: scripts/install [hunk 5]
```diff
@@ -39,7 +25,7 @@ url_regex='^(http://(127\.[0-9]+\.[0-9]+\.[0-9]+|localhost)|https://.*)(:[0-9]+)
 [[ "$redirect_type" = "proxy" ]] && [[ ! $redirect_path =~ $url_regex ]] && ynh_die \
 "For secure reason, you can't use an unencrypted http remote destination couple with ssowat for your reverse proxy: $redirect_path" 1
 
-# Save extra settings
+# Save extra settings
 ynh_app_setting_set --app=$app --key=redirect_type --value=$redirect_type
 ynh_app_setting_set --app=$app --key=redirect_path --value=$redirect_path
```
Comment (alexAubin) [inline scripts/install]:
```suggestion
```

File: scripts/install [hunk 6]
```diff
@@ -76,16 +62,9 @@ ynh_script_progression --message="Configuring permissions..." --weight=2
 if [ "$redirect_type" != "private_proxy" ]
 then
 	# unprotected_uris allows SSO credentials to be passed anyway.
-	ynh_permission_update --permission="main" --add="visitors"
+    ynh_permission_update --permission="main" --add="visitors"
 fi
```
Comment (alexAubin) [inline scripts/install]: hmpf, what do with this ... the proxy type and the ACL should be two different things ..

File: scripts/upgrade [hunk 7]
```diff
@@ -34,11 +23,12 @@ fi
 
 # Migrate away from old stuff with 'is_public' and old redirect type names
 is_public=$(ynh_app_setting_get "$app" is_public)
+
 if [ -n "$is_public" ]
 then
-    if [ "$is_public" = "Yes" ]; then
+     if [ "$is_public" = "Yes" ]; then
```
Comment (alexAubin) [inline scripts/upgrade]: Ogod

**Comment by alexAubin** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

File: tests.toml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+[reverseproxy]
+
+    args.redirect_type = "reverseproxy"
+    args.target = "http://127.0.0.1"
+
+    exclude = ["install.private", "install.multi", "backup_restore"]
+
+    test_upgrade_from.09cf1c6b.name = "v1 era"
+    test_upgrade_from.09cf1c6b.args.domain = "domain.tld"
+    test_upgrade_from.09cf1c6b.args.redirect_type = "proxy_public"
```
Comment (alexAubin) [inline tests.toml]:
```suggestion
    test_upgrade_from.09cf1c6b.args.redirect_type = "proxy"
```

File: conf/nginx-reverseproxy.conf [hunk 9]
```diff
@@ -1,5 +1,7 @@
-location YNH_LOCATION {
-  proxy_pass        YNH_REDIRECT_PATH;
+#sub_path_only rewrite ^__PATH__$ __PATH__/ permanent;
```
Comment (Salamandar) [inline conf/nginx-reverseproxy.conf:1]: Meh, are you sure ?

Maybe the $target needs no /…

**Review by Salamandar — APPROVED** [review]

File: conf/nginx-reverseproxy.conf (hunk 9 above)
Comment (alexAubin) [reply conf/nginx-reverseproxy.conf:1]: zblerg idk, i can't think of any case where that would really make a difference ...

**Comment by alexAubin** [comment]
!testme

**Comment by Salamandar** [comment]
> args.redirect_path = "http://127.0.0.1"

> Running: yunohost app install --no-remove-on-failure --force /app_folder -a "domain=sub.domain.tld&redirect_type=proxy&**redirect_path=/**&path=/redirect&is_public=false"


ace-ventura-spear

**Comment by alexAubin** [comment]
!testme

**Comment by alexAubin** [comment]
!testme

**Pull request merged**